From adde956f4d5df9d06cd6b9bc46e11bc551c2c2db Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Thu, 30 Apr 2020 13:31:30 +0300
Subject: [PATCH 42/46] iot-gate-imx8: add support for the baseboard
 SB-IOTGIMX8

The SB-IOTGIMX8 is the baseboard for the IOT-GATE-iMX8.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 arch/arm/dts/Makefile               |  3 ++-
 arch/arm/dts/sb-iotgimx8.dts        | 20 ++++++++++++++++++++
 configs/ucm-imx8m-mini_d1_defconfig |  4 ++--
 configs/ucm-imx8m-mini_d4_defconfig |  4 ++--
 configs/ucm-imx8m-mini_defconfig    | 10 +++++-----
 include/configs/ucm-imx8m-mini.h    |  4 ++--
 6 files changed, 33 insertions(+), 12 deletions(-)
 create mode 100644 arch/arm/dts/sb-iotgimx8.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 126b4d302c..c9b546bf8f 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -479,7 +479,8 @@ dtb-$(CONFIG_ARCH_IMX8) += fsl-imx8dx-17x17-val.dtb \
 		fsl-imx8qxp-lpddr4-arm2.dtb \
 		fsl-imx8qxp-mek.dtb
 
-dtb-$(CONFIG_TARGET_UCM_IMX8M_MINI) += ucm-imx8m-mini.dtb
+dtb-$(CONFIG_TARGET_UCM_IMX8M_MINI) += ucm-imx8m-mini.dtb \
+		sb-iotgimx8.dtb
 
 dtb-$(CONFIG_RCAR_GEN3) += \
 	r8a7795-h3ulcb.dtb \
diff --git a/arch/arm/dts/sb-iotgimx8.dts b/arch/arm/dts/sb-iotgimx8.dts
new file mode 100644
index 0000000000..99d8b9846f
--- /dev/null
+++ b/arch/arm/dts/sb-iotgimx8.dts
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2020 CompuLab
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "ucm-imx8m-mini.dts"
+
+/ {
+	model = "CompuLab IOT-GATE-iMX8";
+	compatible = "sb-iotgimx8", "cpl,ucm-imx8m-mini", "fsl,imx8mm-evk", "fsl,imx8mm";
+};
diff --git a/configs/ucm-imx8m-mini_d1_defconfig b/configs/ucm-imx8m-mini_d1_defconfig
index 0caccdcf59..e732ce1cd4 100644
--- a/configs/ucm-imx8m-mini_d1_defconfig
+++ b/configs/ucm-imx8m-mini_d1_defconfig
@@ -6,11 +6,11 @@ CONFIG_TARGET_UCM_IMX8M_MINI=y
 CONFIG_RAM_1G=y
 CONFIG_SPL_MMC_SUPPORT=y
 CONFIG_IMX_OPTEE=y
-CONFIG_DEFAULT_DEVICE_TREE="ucm-imx8m-mini"
+CONFIG_DEFAULT_DEVICE_TREE="sb-iotgimx8"
 CONFIG_FIT=y
 CONFIG_SPL_LOAD_FIT=y
 CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=arch/arm/mach-imx/spl_sd.cfg,SPL_TEXT_BASE=0x7E1000"
-CONFIG_DEFAULT_FDT_FILE="ucm-imx8m-mini.dtb"
+CONFIG_DEFAULT_FDT_FILE="sb-iotgimx8.dtb"
 CONFIG_ARCH_MISC_INIT=y
 CONFIG_SPL=y
 CONFIG_SPL_BOARD_INIT=y
diff --git a/configs/ucm-imx8m-mini_d4_defconfig b/configs/ucm-imx8m-mini_d4_defconfig
index a73b83c6cd..576460a56a 100644
--- a/configs/ucm-imx8m-mini_d4_defconfig
+++ b/configs/ucm-imx8m-mini_d4_defconfig
@@ -6,11 +6,11 @@ CONFIG_TARGET_UCM_IMX8M_MINI=y
 CONFIG_RAM_4G=y
 CONFIG_SPL_MMC_SUPPORT=y
 CONFIG_IMX_OPTEE=y
-CONFIG_DEFAULT_DEVICE_TREE="ucm-imx8m-mini"
+CONFIG_DEFAULT_DEVICE_TREE="sb-iotgimx8"
 CONFIG_FIT=y
 CONFIG_SPL_LOAD_FIT=y
 CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=arch/arm/mach-imx/spl_sd.cfg,SPL_TEXT_BASE=0x7E1000"
-CONFIG_DEFAULT_FDT_FILE="ucm-imx8m-mini.dtb"
+CONFIG_DEFAULT_FDT_FILE="sb-iotgimx8.dtb"
 CONFIG_ARCH_MISC_INIT=y
 CONFIG_SPL=y
 CONFIG_SPL_BOARD_INIT=y
diff --git a/configs/ucm-imx8m-mini_defconfig b/configs/ucm-imx8m-mini_defconfig
index 2955b2f907..3dbed4f9d1 100644
--- a/configs/ucm-imx8m-mini_defconfig
+++ b/configs/ucm-imx8m-mini_defconfig
@@ -1,15 +1,16 @@
 CONFIG_ARM=y
 CONFIG_ARCH_IMX8M=y
 CONFIG_SYS_TEXT_BASE=0x40200000
+CONFIG_SPL_GPIO_SUPPORT=y
 CONFIG_SYS_MALLOC_F_LEN=0x2000
 CONFIG_TARGET_UCM_IMX8M_MINI=y
 CONFIG_SPL_MMC_SUPPORT=y
 CONFIG_IMX_OPTEE=y
-CONFIG_DEFAULT_DEVICE_TREE="ucm-imx8m-mini"
+CONFIG_DEFAULT_DEVICE_TREE="sb-iotgimx8"
 CONFIG_FIT=y
 CONFIG_SPL_LOAD_FIT=y
 CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=arch/arm/mach-imx/spl_sd.cfg,SPL_TEXT_BASE=0x7E1000"
-CONFIG_DEFAULT_FDT_FILE="ucm-imx8m-mini.dtb"
+CONFIG_DEFAULT_FDT_FILE="sb-iotgimx8.dtb"
 CONFIG_ARCH_MISC_INIT=y
 CONFIG_SPL=y
 CONFIG_SPL_BOARD_INIT=y
@@ -50,8 +51,6 @@ CONFIG_DM_SPI_FLASH=y
 CONFIG_SPI_FLASH=y
 CONFIG_SPI_FLASH_STMICRO=y
 CONFIG_DM_ETH=y
-CONFIG_USB_HOST_ETHER=y
-CONFIG_USB_ETHER_ASIX88179=y
 CONFIG_PINCTRL=y
 CONFIG_PINCTRL_IMX8M=y
 CONFIG_DM_REGULATOR=y
@@ -67,4 +66,5 @@ CONFIG_USB_GADGET_MANUFACTURER="FSL"
 CONFIG_USB_GADGET_VENDOR_NUM=0x0525
 CONFIG_USB_GADGET_PRODUCT_NUM=0xa4a5
 CONFIG_SDP_LOADADDR=0x40400000
-CONFIG_SPL_GPIO_SUPPORT=y
+CONFIG_USB_HOST_ETHER=y
+CONFIG_USB_ETHER_ASIX88179=y
diff --git a/include/configs/ucm-imx8m-mini.h b/include/configs/ucm-imx8m-mini.h
index cd1ed129f9..3fa32c72d2 100644
--- a/include/configs/ucm-imx8m-mini.h
+++ b/include/configs/ucm-imx8m-mini.h
@@ -127,7 +127,7 @@
 	"fdt_addr=0x43000000\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fdt=yes\0" \
-	"fdt_file="CONFIG_DEFAULT_DEVICE_TREE".dtb\0" \
+	"fdt_file=sb-iotgimx8-can.dtb\0" \
 	"initrd_addr=0x43800000\0"		\
 	"initrd_high=0xffffffffffffffff\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
@@ -222,7 +222,7 @@
 
 /* Monitor Command Prompt */
 #undef CONFIG_SYS_PROMPT
-#define CONFIG_SYS_PROMPT		"u-boot=> "
+#define CONFIG_SYS_PROMPT		"IOT-GATE-iMX8 => "
 #define CONFIG_SYS_PROMPT_HUSH_PS2     "> "
 #define CONFIG_SYS_CBSIZE              2048
 #define CONFIG_SYS_MAXARGS             64
-- 
2.11.0

